alternator/executor.cc

```diff
@@ -2226,6 +2226,7 @@ future<executor::request_return_type> executor::batch_write_item(client_state& c
                     co_return api_error::validation("Provided list of item keys contains duplicates");
                 }
                 used_keys.insert(std::move(mut_key));
+                batch_size++;
```

Comment:
nit: can't we just use `batch_size += used_keys.size()` after we exit the inner loop adding the items?

Reply:
It will not have the same all-or-nothing effect incase of validation error

Reply:
@amnon, I don't understand - when there's a validation error, NO item gets written or read at all (the whole operation is ignored), so we don't want to increase the counter at all. Do you do that? It will be nice to have a test for that as well.

Reply:
That's what I wrote; with the current implementation, when there is a validation error, we do not increase the batch_size. This functionality will be lost if I follow @mykaul suggestion

Reply:
I don't see why. The addition (using the .size() function) should be post-validation.

Reply:
it will be inside the outer loop, so it can still fail mid-way

Reply:
I don't follow... No matter how you count the items, you only add the count to a local variable, and the global counter is only incremented by the count after all the validation and counting is over.